xarray/backends/plugins.py

```diff
@@ -111,10 +111,17 @@ def guess_engine(store_spec):
 
 
 def get_backend(engine):
-    """Select open_dataset method based on current engine"""
-    engines = list_engines()
-    if engine not in engines:
-        raise ValueError(
-            f"unrecognized engine {engine} must be one of: {list(engines)}"
-        )
-    return engines[engine]
+    """Select open_dataset method based on current engine."""
+    if isinstance(engine, str):
+        engines = list_engines()
+        if engine not in engines:
+            raise ValueError(
+                f"unrecognized engine {engine} must be one of: {list(engines)}"
+            )
+        backend = engines[engine]
+    elif issubclass(type(engine), BackendEntrypoint):
```

Comment:
`type(engine)` is `type`:
```suggestion
    elif issubclass(engine, BackendEntrypoint):
```

Reply:
This crashes if engine="2" or engine=2. That's why I use type().

Reply:
ah, I missed that. What about
```suggestion
    elif isinstance(engine, type) and issubclass(engine, BackendEntrypoint):
```

Reply:
`inspect` was already imported here so I was thinking `inspect.isclass(engine) and issubclass(engine, BackendEntrypoint)`.

Reply:
Done.